src/Http/Routing/src/Matching/HttpMethodDictionaryPolicyJumpTable.cs

```diff
@@ -7,11 +7,104 @@ namespace Microsoft.AspNetCore.Routing.Matching;
 
 internal sealed class HttpMethodDictionaryPolicyJumpTable : PolicyJumpTable
 {
+    internal struct HttpMethodsJumpTable
```

Comment:
It would be nice if the struct could be readonly. Could the dictionary of destinations be passed into the ctor? The values could be set in the ctor, the struct made readonly, and struct fields readonly.

Reply:
Alternatively, this could be a class. Did you make it a struct to avoid the allocation? Jump tables probably last the lifetime of an app. Allocations aren't a concern.

Reply:
If we go all the way with optimizing this - https://github.com/dotnet/aspnetcore/pull/51803/files#r1382732662 - then this type is probably better as a class.

Reply:
Yes I made it a struct because I did not want allocation. I can make it a class or a readonly struct too, as I see you prefer it as a class rather.